producer-js/browser-based-ingestion/library/src/putMediaCall.js

```diff
@@ -0,0 +1,45 @@
+const AWS = require('aws-sdk')
+const AWS4 = require('aws4')
+
+const putMedia = async function (videoFile, service, region, accessKeyID, secretAccessKey, sessionToken, dataEndpoint, streamName) {
+  const dataEndpointPutMedia = new AWS.Endpoint(dataEndpoint)
+  const url = dataEndpointPutMedia.href + 'putMedia'
+
+  const options = AWS4.sign(
+    {
+      method: 'POST',
+      host: dataEndpointPutMedia.host,
+      path: '/putMedia',
+      service: service,
+      region: region,
+      headers: {
+        Accept: '*/*',
+        'x-amzn-fragment-timecode-type': 'RELATIVE',
+        'x-amzn-producer-start-timestamp': Date.now() / 1000,
+        'x-amzn-stream-name': streamName
+      }
+    },
+    {
+      accessKeyId: accessKeyID,
+      secretAccessKey: secretAccessKey,
+      sessionToken: sessionToken
+    }
+  )
+  options.body = videoFile
+  const request = new Request(url, options)
+  fetch(request)
+    .then(function (response) {
+      return response.text()
+    })
+    .then(function (data) {
+      console.log(data)
+      if (data.includes('"EventType":"ERROR"')) {
+        alert('The following error has occurred:\n' + data + '\nReload the page and try again')
+      }
+    })
+    .catch(function (error) {
```

Comment:
No need to catch this, since the higher level should catch it and handle the UI part.